app/src/main/java/ch/epfl/sdp/drone3d/service/impl/location/AndroidLocationService.kt

```diff
@@ -42,22 +42,20 @@ class AndroidLocationService @Inject constructor(
         return currentPermission == PackageManager.PERMISSION_GRANTED && getMyLocationProvider() != null
     }
 
+    // Permission is checked on isLocationEnabled()
+    @SuppressLint("MissingPermission")
     override fun getCurrentLocation(): LatLng? {
```

Comment:
We can remove this by just adding a small if permission is not granted then return null. (Or ask for permission)

What I was saying was not to remove the edge case where the permission was not granted, rather to clean the try/catch and remove it as it would have the exact same behavior.

Reply:
I missed a line